pentaho-gwt-widgets/src/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java

```diff
@@ -148,6 +148,8 @@ public String getLabelTooltip() {
   }
 
   protected void closeTab() {
+    String frameId = getContent().getElement().getElementsByTagName( "iframe" ).getItem( 0 ).getAttribute( "id" );
```

Comment:
Please program that in a more defensive way. What happens when you have (for some reason) no iframe in your tab? The DOM is a shared resource and anyone (including badly written scripts written by 3rd parties) can modify it freely.

Check whether the nodelist returned by getElementsByTagName is empty. If you were to access the first element in that case, you'd get an error. If empty, neither try to get the ID nor fire the event.